dpiprobe.go

```diff
@@ -1,30 +1,48 @@
 package main
 
 import (
+	"encoding/binary"
 	"encoding/hex"
 	"errors"
 	"flag"
 	"fmt"
-	"github.com/google/gopacket"
-	"github.com/google/gopacket/layers"
-	"github.com/google/gopacket/pcap"
-	"golang.org/x/net/idna"
-	"golang.org/x/net/ipv4"
 	"math/big"
 	"math/rand"
 	"net"
 	"os"
 	"strings"
 	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"github.com/refraction-networking/utls"
+	"golang.org/x/net/idna"
+	"golang.org/x/net/ipv4"
 )
 
 func main() {
 	maxTtl := flag.Uint("ttl", 30, "Maximum number of hops.")
-	tcpSynTrace := flag.Bool("syn", false, "Force TCP SYN trace.")
+	connectionMode := flag.String("mode", "", "Connection mode: (syn|http|https).")
 	disableIpPtrLookup := flag.Bool("n", false, "Disable IP PTR lookup.")
 	timeoutSeconds := flag.Uint("t", 15, "Timeout for each hop.")
+	port := flag.Uint("port", 0, "Port number.")
 	flag.Parse()
 
+	switch *connectionMode {
+	case "http", "syn":
+		if *port == 0 {
+			*port = 80
+		}
+	case "https":
+		if *port == 0 {
+			*port = 443
+		}
+	default:
+		fmt.Printf("Invalid mode (%s) \nRun \"dpiprobe --help\" for usage instructions.\n", *connectionMode)
```

Comment:
In case the connection mode is not specified, choose a default based on the port. Add a case for the empty value. If the port was specified as 80, default to http, if the port was specified as 443, default to https, if anything else default to syn.

Reply:
I will work on that